creator-node/src/services/stateMachineManager/stateReconciliation/recoverOrphanedData.jobProcessor.ts

```diff
@@ -142,67 +137,146 @@ const _storeWalletsWithThisNodeInReplica = async (
+      // recovery will short circuit later to avoid wiping state on any node in the user's replica set.
+    }
   } while (batchOfUsers?.length === NUM_USERS_PER_QUERY && prevUserId !== 0)
 
   // Save the wallets as a redis set and return the set cardinality
-  const numWalletsWithNodeInReplicaSet = await redisClient.sadd(
-    WALLETS_WITH_NODE_IN_REPLICA_SET_KEY,
-    walletsWithNodeInReplicaSetArr
-  )
+  const numWalletsWithNodeInReplicaSet: number =
+    walletsWithNodeInReplicaSetArr.length
+      ? await redisClient.sadd(
+          WALLETS_WITH_NODE_IN_REPLICA_SET_KEY,
+          walletsWithNodeInReplicaSetArr
+        )
+      : 0
   return numWalletsWithNodeInReplicaSet
 }
 
 /**
- * Gets a sync request that can be issued to "force wipe" (modified forceResync=true flag):
- *  1. Merges orphaned data from this node into the user's primary.
- *  2. Wipes the user's data from this node.
- *  3. *DON'T* resync from the primary to this node -- this is what non-modified forceResync would do.
- * @param wallet the wallet of the user with data orphaned on this node
+ * Finds wallets that are orphaned and adds them to a redis set.
+ * (Set of orphaned wallets) = (set of wallets on this node) - (set of wallets with this node in their replica set)
+ * @returns number of wallets that have data orphaned on this node
  */
-const _getSyncReqToRecoverOrphanedData = async (
-  wallet: string,
+const _saveWalletsWithOrphanedDataToRedis = async () => {
+  const numWalletsOrphaned: number = await redisClient.sdiffstore(
+    WALLETS_ORPHANED_KEY,
+    WALLETS_ON_NODE_KEY,
+    WALLETS_WITH_NODE_IN_REPLICA_SET_KEY
+  )
+  return numWalletsOrphaned
+}
+
+/**
+ * Pops from redis set in batches and issues requests to move data that's orphaned on this node back to each user's primary.
+ * Each request will sync from this node to a primary and then "force wipe" this node:
+ *  1. Merge orphaned data from this node into the user's primary.
+ *  2. Wipe the user's data from this node.
+ *  3. *DON'T* resync from the primary to this node -- this is what non-modified forceResync would do -- and instead set forceWipe=true.
+ * @param {number} numWalletsWithOrphanedData number of users to issue requests to recover orphaned data for
+ * @param {string} discoveryNodeEndpoint the endpoint of a discovery node to make queries to
+ * @param {Logger} logger logger
+ * @return {number} number of requests successfully issued
+ */
+const _batchIssueReqsToRecoverOrphanedData = async (
+  numWalletsWithOrphanedData: number,
+  discoveryNodeEndpoint: string,
   logger: Logger
-): Promise<IssueSyncRequestJobParams | undefined> => {
-  const primaryEndpoint = await _getPrimaryForWallet(wallet, logger)
-  if (!primaryEndpoint) return undefined
+): Promise<number> => {
+  let requestsIssued = 0
+  for (
+    let i = 0;
+    i < numWalletsWithOrphanedData;
+    i += NUM_USERS_TO_RECOVER_PER_BATCH
+  ) {
+    const walletsWithOrphanedData = await redisClient.spop(
```

Comment:
whats intent here - reduce redis calls?

Reply:
yeah to reduce redis calls and keep the set in redis instead of loading it all into memory like the previous iteration. seems like better practice

Reply:
sounds good!